Tools/machines/karolina-it4i/install_gpu_dependencies.sh

```diff
@@ -0,0 +1,138 @@
+  git checkout v1.21.1
+  cd -
+else
+  git clone -b v1.21.1 https://github.com/Blosc/c-blosc.git $HOME/src/c-blosc
+fi
+rm -rf $HOME/src/c-blosc-gpu-build
+cmake -S $HOME/src/c-blosc -B $HOME/src/c-blosc-gpu-build -DBUILD_TESTS=OFF -DBUILD_BENCHMARKS=OFF -DDEACTIVATE_AVX2=OFF -DCMAKE_INSTALL_PREFIX=${SW_DIR}/c-blosc-1.21.1
+cmake --build $HOME/src/c-blosc-gpu-build --target install --parallel 16
+rm -rf $HOME/src/c-blosc-gpu-build
+
+# HDF5
+if [ -d $HOME/src/hdf5 ]
+then
+  cd $HOME/src/hdf5
+  git fetch
+  git checkout hdf5-1_14_1-2
+  cd -
+else
+  git clone -b hdf5-1_14_1-2 https://github.com/HDFGroup/hdf5.git src/hdf5
+fi
+rm -rf $HOME/src/hdf5-build
+cmake -S $HOME/src/hdf5 -B $HOME/src/hdf5-build -DBUILD_TESTING=OFF -DHDF5_ENABLE_PARALLEL=ON -DCMAKE_INSTALL_PREFIX=${SW_DIR}/hdf5-1.14.1.2
+cmake --build $HOME/src/hdf5-build --target install --parallel 16
+rm -rf $HOME/src/hdf5-build
+
+# ADIOS2
+if [ -d $HOME/src/adios2 ]
+then
+  cd $HOME/src/adios2
+  git fetch
+  git checkout v2.8.3
+  cd -
+else
+  git clone -b v2.8.3 https://github.com/ornladios/ADIOS2.git $HOME/src/adios2
+fi
+rm -rf $HOME/src/adios2-gpu-build
+cmake -S $HOME/src/adios2 -B $HOME/src/adios2-gpu-build -DADIOS2_USE_Blosc=ON -DADIOS2_USE_HDF5=OFF -DADIOS2_USE_Fortran=OFF -DADIOS2_USE_Python=OFF -DADIOS2_USE_ZeroMQ=OFF -DCMAKE_INSTALL_PREFIX=${SW_DIR}/adios2-2.8.3
+cmake --build $HOME/src/adios2-gpu-build --target install -j 16
+rm -rf $HOME/src/adios2-gpu-build
+
+# BLAS++ (for PSATD+RZ)
+if [ -d $HOME/src/blaspp ]
+then
+  cd $HOME/src/blaspp
+  git fetch
+  git checkout master
+  git pull
+  cd -
+else
+  git clone https://github.com/icl-utk-edu/blaspp.git $HOME/src/blaspp
+fi
+rm -rf $HOME/src/blaspp-gpu-build
+cmake -S $HOME/src/blaspp -B $HOME/src/blaspp-gpu-build -Duse_openmp=OFF -Dgpu_backend=cuda -DCMAKE_CXX_STANDARD=17 -DCMAKE_INSTALL_PREFIX=${SW_DIR}/blaspp-master
+cmake --build $HOME/src/blaspp-gpu-build --target install --parallel 16
+rm -rf $HOME/src/blaspp-gpu-build
+
+# LAPACK++ (for PSATD+RZ)
+if [ -d $HOME/src/lapackpp ]
+then
+  cd $HOME/src/lapackpp
+  git fetch
+  git checkout master
+  git pull
+  cd -
+else
+  git clone https://github.com/icl-utk-edu/lapackpp.git $HOME/src/lapackpp
+fi
+rm -rf $HOME/src/lapackpp-gpu-build
+CXXFLAGS="-DLAPACK_FORTRAN_ADD_" cmake -S $HOME/src/lapackpp -B $HOME/src/lapackpp-gpu-build -DCMAKE_CXX_STANDARD=17 -Dbuild_tests=OFF -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON -DCMAKE_INSTALL_PREFIX=${SW_DIR}/lapackpp-master
+cmake --build $HOME/src/lapackpp-gpu-build --target install --parallel 16
+rm -rf $HOME/src/lapackpp-gpu-build
+
+
+# Python ######################################################################
+#
+python3 -m pip install --upgrade pip
+python3 -m pip install --upgrade virtualenv
+python3 -m pip cache purge
+rm -rf ${SW_DIR}/venvs/warpx-gpu
+python3 -m venv ${SW_DIR}/venvs/warpx-gpu
+source ${SW_DIR}/venvs/warpx-gpu/bin/activate
+python3 -m pip install --upgrade pip
+python3 -m pip install --upgrade wheel
```

Comment:
@berceanu your error tells me that this line was not executed or your vent is not active

Reply:
Yeah so what happened was this: I was sourcing the `WarpX` profile at login (from my `.bashrc`), but at that point the venv did not exist. I then installed all the dependencies, including the venv, but forgot to source the profile again before attempting to compile `warpx`. Should we add this to the instructions? One need to either log out and back in or manually source the profile once more in between these 2 steps.

Reply:
Great :+1: 

Oh good point, one has to activate the env after install of the dependencies generally. I will improve the workflow to avoid a login/out at that point.

Reply:
Fixed here and in #4029